internal/pkg/crypto/ca.go

```diff
@@ -18,28 +18,28 @@ type CA struct {
 	CaPrivateKey *rsa.PrivateKey
 }
 
-func (ca *CA) PEMEncodeCertificate() (string, error) {
+func (ca *CA) PEMEncodeCertificate() (*bytes.Buffer, error) {
```

Comment:
I think we should keep returning encoded PEM as strings, given that in it we get base64 payload, ascii compatible, so utf compatible. Hence why they have been encoded.

Reply:
To void another conversion. This function is returning `*bytes.Buffer` because I do not need string value. All the usage of the data returned by this function needs to be bytes. Not strings. But, I keep using `*bytes.Buffer` instead of `[]byte` because if in the next changes is necessary a string I could access it using the available function [`String`](https://pkg.go.dev/bytes@go1.21.4#Buffer.String) or [`Bytes`](https://pkg.go.dev/bytes@go1.21.4#Buffer.Bytes).

Reply:
I still think that String would be a better fit than bytes.Buffer or []Bytes. Once we have created the PEM encoded certificate, we don't need to change it (String is immutable, the other not), we don't need to sequentially add chars (so we don't need a Buffer that is cost-effective for that), and the result is supposed to be human-readable (String).